### PR TITLE
Update logging LogRecord msg attribute

### PR DESCRIPTION
**Pull request by aminalaee**
Closes https://github.com/python/typeshed/issues/9913

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Looks like this is a partial revert of #1776

**Comment by AlexWaygood** [comment]
In the issue, you wrote:

> According to the Python docs for the `LogRecord` in `logging` module the `msg` argument and property is a string:
> https://docs.python.org/3/library/logging.html#logging.LogRecord

However, the CPython docs are pretty inconsistent here: later on down in the docs, it's [specifically stated]() that it's okay to pass non-`str` objects in as messages:

> In the preceding sections and examples, it has been assumed that the message passed when logging the event is a string. However, this is not the only possibility. You can pass an arbitrary object as a message, and its [`__str__()`](https://docs.python.org/3/reference/datamodel.html#object.__str__) method will be called when the logging system needs to convert it to a string representation. In fact, if you want to, you can avoid computing a string representation altogether - for example, the [SocketHandler](https://docs.python.org/3/library/logging.handlers.html#logging.handlers.SocketHandler) emits an event by pickling it and sending it over the wire.

This seems to me like a bug in the CPython docs: they shouldn't say one thing up at the top, only to say a different thing later down.

**Comment by aminalaee** [comment]
I hadn't really checked the other places, you're right.
This is much deeper in the docs than the CPython and typeshed. I guess we can close this and I can create a PR on the CPython docs.

**Review by AlexWaygood — CHANGES_REQUESTED** [review]
Since the CPython docs specifically say that it's okay to pass in arbitrary non-`str` objects to the `msg` parameter, I think a better fix would be to keep `object` for the parameter annotation here, but change this attribute annotation so that it is also `object`:

https://github.com/python/typeshed/blob/dd2818a41db6cd31e4680abf5e1362a7e5bfb5a6/stdlib/logging/__init__.pyi#L392

**Review by AlexWaygood — APPROVED** [review]
Thank you! A docs PR over at CPython would also be great. If you ping me on the PR, I can help review it :)

File: stdlib/logging/__init__.pyi [hunk 1]
```diff
@@ -389,7 +389,7 @@ class LogRecord:
     msecs: float
     # Only created when logging.Formatter.format is called. See #6132.
     message: str
-    msg: str
+    msg: object
```
[the remark this answers is not in the document]
Comment (aminalaee) [reply stdlib/logging/__init__.pyi]: This will avoid unnecessary effect on those projects I guess. Cool/

**Pull request merged**